pkg/generate/app/app.go

```diff
@@ -434,6 +436,11 @@ func (r *DeploymentConfigRef) DeploymentConfig() (*deployapi.DeploymentConfig, e
 	selector := map[string]string{
 		"deploymentconfig": r.Name,
 	}
+	if len(r.Labels) > 0 {
+		if err := util.MergeInto(selector, r.Labels, util.ErrorOnDifferentDstKeyValue); err != nil {
```

Comment:
I don't think you want to error here.  If someone provides deploymentconfig=foo, they probably want that behavior.  Doesn't seem accidental.